File: API LavaRapido/model/Fornecedor.py
from pydantic import BaseModel
from data.database import Base
from sqlalchemy import Column, VARCHAR, INTEGER

class Fornecedor(Base):
    __tablename__ = "tb_fornecedor"

    id:  int = Column(INTEGER, primary_key=True, autoincrement=True)
    nome: str = Column(VARCHAR(100), nullable=False)
    cnpj: str = Column(VARCHAR(100), nullable=False)
    telefone: str = Column(VARCHAR(100), nullable=False)
    email: str = Column(VARCHAR(100), nullable=False)
    cep: str = Column(VARCHAR(100), nullable=False)
    logradouro: str = Column(VARCHAR(255), nullable=False)
    bairro: str = Column(VARCHAR(100), nullable=False)
    numero: int = Column(INTEGER,  nullable=False)
    complemento: str = Column(VARCHAR(100), nullable=True)
    cidade: str = Column(VARCHAR(100), nullable=False)
    estado: str = Column(VARCHAR(100), nullable=False)


class FornecedorBase(BaseModel):
    nome: str
    cnpj: str
    telefone: str
    email: str
    cep: str
    logradouro: str
    bairro: str
    numero: int
    complemento: str
    cidade: str
    estado: str


class FornecedorRequest(FornecedorBase):
    ...


class FornecedorResponse(FornecedorBase):
    id: int

    class Config:
        from_attributes = True
        populate_by_name = True

File: lava_rapido_final/lib/model/fornecedor.dart
class Fornecedor {
  int id;
  String nome;
  String cnpj;
  String telefone;
  String email;
  String cep;
  String logradouro;
  String bairro;
  int numero;
  String complemento;
  String cidade;
  String estado;

  Fornecedor(
      this.id,
      this.nome,
      this.cnpj,
      this.telefone,
      this.email,
      this.cep,
      this.logradouro,
      this.bairro,
      this.numero,
      this.complemento,
      this.cidade,
      this.estado);

  factory Fornecedor.fromJson(Map<String, dynamic> json) {
    return Fornecedor(
      json['id'],
      json['nome'],
      json['cnpj'],
      json['telefone'],
      json['email'],
      json['cep'],
      json['logradouro'],
      json['bairro'],
      json['numero'],
      json['complemento'],
      json['cidade'],
      json['estado'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'cnpj': cnpj,
      'telefone': telefone,
      'email': email,
      'cep': cep,
      'logradouro': logradouro,
      'bairro': bairro,
      'numero': numero,
      'complemento': complemento,
      'cidade': cidade,
      'estado': estado,
    };
  }
}
